File: problem-3/main.cpp
#include <iostream>
int largest_prime_factor(long int n){
    int i = 2;
    while (i * i <= n) {
        if (n % i != 0)
        {
            i++;
        }
        else {
            n /= i;
        } 
    }
    return n;
    
}
int main(){
    std::cout << largest_prime_factor(600851475143);
}

File: problem-3/main.go
package main
import "fmt"
func main(){
	fmt.Println(largest_prime_factor(600851475143))
}
func largest_prime_factor(n uint64) uint64 {
	var i uint64 = 2
	for i*i <= n {
		if n % i != 0 {
			i++
		} else {
			n /= i
		}
	}
	return n
}